platform.py: fix incorrect bestDiskLabelType

The comparison is intended to find a label type that can support the
size of the disk, but it has the logic reversed

Figured-out-by: Don Penney <Don.Penney@windriver.com>

Upstream-Status: Pending
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/platform.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/platform.py b/pyanaconda/platform.py
index a3d1154..8bae425 100644
--- a/pyanaconda/platform.py
+++ b/pyanaconda/platform.py
@@ -32,6 +32,9 @@ import gettext
 _ = lambda x: gettext.ldgettext("anaconda", x)
 N_ = lambda x: x
 
+import logging
+log = logging.getLogger("storage")
+
 class Platform(object):
     """Platform
 
@@ -105,10 +108,11 @@ class Platform(object):
             labelType = self.defaultDiskLabelType
             for lt in self.diskLabelTypes:
                 l = parted.freshDisk(device=device.partedDevice, ty=lt)
-                if l.maxPartitionStartSector < device.partedDevice.length:
+                if l.maxPartitionStartSector > device.partedDevice.length:
                     labelType = lt
                     break
 
+        log.debug("device %s, bestDiskLabelType %s" % (device, labelType))
         return labelType
 
     @property
-- 
1.9.1

